src/main/java/com/coveo/Check.java

```diff
@@ -1,13 +1,55 @@
 package com.coveo;
 
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
 
 @Mojo(name = "check", defaultPhase = LifecyclePhase.VERIFY)
 public class Check extends AbstractFMT {
 
+  @Parameter(defaultValue = "true", property = "displayFiles")
+  private boolean displayFiles;
+
+  @Parameter(defaultValue = "100", property = "displayLimit")
+  private int displayLimit;
+
+  protected List<String> filesNotFormatted = new ArrayList<String>();
+
   @Override
   protected boolean isValidateOnly() {
     return true;
   }
+
+  /** Hook called when a file is not compliant. */
+  protected void nonComplyingFile(File file) {
+    if (isValidateOnly()) {
+      filesNotFormatted.add(file.getAbsolutePath());
+    }
+  }
+
+  /** Display some statistics on the files that are not complying */
+  protected void additionalFailComplying() {
+    if (isValidateOnly() && nonComplyingFiles > 0) {
+
+      // do not support limit < 1
+      displayLimit = max(1, displayLimit);
+
+      // Display first displayLimit files not formatted
+      if (displayFiles) {
+        if (nonComplyingFiles > displayLimit) {
```

Comment:
I think it would be cleaner if you printed all `displayLimit` rows and then at the end printed something like `... and ${displayLimit - nonComplyingFiles} more`.

AKA I think it's much more likely that people will spot it at the end then at the start.